src/Microsoft.Health.Dicom.SqlServer/Features/Store/SqlIndexDataStore.cs

```diff
@@ -48,6 +56,16 @@ public async Task<long> CreateInstanceIndexAsync(DicomDataset instance, Cancella
             using (SqlConnectionWrapper sqlConnectionWrapper = await _sqlConnectionFactoryWrapper.ObtainSqlConnectionWrapperAsync(cancellationToken))
             using (SqlCommandWrapper sqlCommandWrapper = sqlConnectionWrapper.CreateSqlCommand())
             {
+                IEnumerable<InsertStringCustomTagTableTypeV1Row> stringCustomTags =
+                    DetermineKeysAndTagsForDataType(storedCustomTagEntries, CustomTagDataType.StringData).Select(x => new InsertStringCustomTagTableTypeV1Row(x.Key, instance.GetString(x.Value)));
```

Comment:
what happens when there is no value for that key in the dicomdataset? do we insert null? we should avoid that.